src/utils/eventValidator.js

```
@@ -12,6 +12,36 @@ const VALID_TRIGGERS = ['post_gig', 'travel', 'random']
+  }
+  if (effect.type === 'composite') {
+    if (!Array.isArray(effect.effects) || effect.effects.length === 0) {
+      throw new Error(
+        `Composite effect must have a non-empty effects array at index ${idx} for event ${eventId}`
+      )
+    }
+    effect.effects.forEach((childEffect, childIdx) => {
+      try {
+        validateEffect(childEffect, eventId, idx)
+      } catch (err) {
+        throw new Error(`Invalid child effect at composite index ${childIdx} for option index ${idx} in event ${eventId}: ${err.message}`)
```

ESLint

[error] There is no `cause` attached to the symptom error being thrown.

(preserve-caught-error)